scripts/setup-macos.sh

```diff
@@ -102,7 +102,18 @@ function update_brew {
 function install_build_prerequisites {
   for pkg in ${MACOS_DEPS}
   do
-    brew install --formula $pkg && echo "Installation of $pkg is successful" || brew upgrade --formula $pkg
+    if [[ "${pkg}" =~ ^([0-9a-z-]*):([0-9](\.[0-9\])*)$ ]];
+    then
+      pkg=${BASH_REMATCH[1]}
```

Comment:
The few times I did these steps manually I also had to explicitly unlink the older version (which could have got installed before). Maybe brew didn't unlink if its default version was still available. If that is too detailed to do in this logic, maybe leave a comment for it.

Reply:
@aditi-pandit in our case, we are installing to a completely new directory (~/deps) so there is no existing to unlink.